### ISSUE TITLE
Hook overloaded dalvik function with complex parameter

### ISSUE THREAD

**Issue by Darkkey** [frida/frida-gum#90]
How to correctly instrument f function from c class if it declared as:

    private void f(List<String> l){ .. }
    private void f(boolean b){ .. }

Overload for boolean (c.f.overload("boolean").implementation = .. ) works, but if I want to hook first function, either with

    var C = Dalvik.use("c");
    c.f.overload("java.util.List<java.lang.String>").implementation = function(param){
     ...
    }

or with 

    c.f.overload("java.util.List").implementation = function(param){
     ...
    }

then I'm getting "Error: f: specified argument types do not match any overload".

Thanks.

**Comment by marc1006** [comment]
It's possible to hook it, but at the moment not in this way.

Try
```
c.f.overload("java.lang.Object").implementation = function(param){
  ...
}
```
In the next future we have to implement this function in a proper way
https://github.com/frida/frida-gum/blob/master/bindings/gumjs/gumscript-runtime-dalvik.js#L1829-L1837

**Comment by Darkkey** [comment]
Still not working on frida 4.3.1:

On "java.util.List" I'm getting "Error: f: specified argument types do not match any overload"

If I use "java.lang.Object", then frida frizes along with application on this.f(param);

**Comment by marc1006** [comment]
It should already work with Frida 4.3.1.

I've just created a demo and it works for me
``` java
private void f(boolean b) {
    Toast.makeText(getApplicationContext(), String.valueOf(b), Toast.LENGTH_SHORT).show();
}

private void f(List<String> l) {
    Toast.makeText(getApplicationContext(), String.valueOf(l.size()), Toast.LENGTH_SHORT).show();
}
```

``` javascript
klass.f.overload("java.util.List").implementation = function() {
    this.f.overload("java.util.List").apply(this, arguments);
};

klass.f.overload("boolean").implementation = function() {
    this.f.overload("boolean").apply(this, arguments);
};
```

This works as well (in this case)
``` javascript
klass.f.overload("java.util.List").implementation = function () {
    this.f(arguments[0]);
};

klass.f.overload("boolean").implementation = function () {
    this.f(arguments[0]);
};
```

**Comment by Darkkey** [comment]
Checked -- it works on test sample, but freezes the real application. I think this is an application issue (strange mix of Dalvik and native jni code). Thanks!

**Comment by oleavr** [comment]
Let's assume this is fixed in newly released 4.5.1.
